LEAF_Request_Portal/admin/templates/mod_workflow.tpl

```diff
@@ -187,6 +187,305 @@ function groupListContent(groups) {
+        }
+    });
+
+    dialog.setTitle('Email Reminder');
+    let output = '<label for="edit_email_check">Automated Emails:</label> <input type="checkbox" id="edit_email_check" onclick="editEmailChecked()"><div id="edit_email_container"></div><br>';
+    dialog.setContent(output);
+    dialog.setSaveHandler(function() {
+
+        let seriesData = {
+            'Automate Email Group': $('#edit_email_check').prop('checked'),
+            'Date Selected': $('#edit_dates_selected').val(),
+            'Days Selected': $('#edit_dates_days').val()
+        }
+
+        $.ajax({
+            type: 'POST',
+            data: {
+                CSRFToken: CSRFToken,
+                seriesData: seriesData
+            },
+            url: '../api/workflow/stepdata/' + stepID,
+            success: function (res) {
+                if (res == 1) {
+                    //loadWorkflow(currentWorkflow);
+                    dialog.hide();
+                }
+                else {
+                    alert(res);
+                }
+            },
+            error: function(){ console.log('Failed to save automated email reminder data'); }
+        });
+    });
+
+    dialog.show();
+
+    // dialog changes if data is setup properly
+    if(workflowStep?.stepData !== null) {
+        let stepParse = JSON.parse(workflowStep.stepData);
+        let automateEmailGroup = stepParse.AutomateEmailGroup;
+        let dateSelected = stepParse.DateSelected;
+        let daysSelected = stepParse.DaysSelected;
+
+        if (automateEmailGroup == "true") {
+            $('#edit_email_check').prop('checked', true);
+            editEmailChecked();
+            $("#edit_dates_selected").val(dateSelected);
+            editEmailDateSelected();
+            $("#edit_dates_days").val(daysSelected);
+
+        } else {
+            $('#edit_email_check').prop('checked', false);
+        }
+    }
+}
+
+///// Creating Automated Emails
+function emailChecked() {
+    let emailChecked = document.getElementById("email_check");
+    if (emailChecked.checked) {
+        let selectdatesString = "";
+        selectdatesString += "<br>";
+        selectdatesString += '<select id="dates_selected" onchange="emailDateSelected()">';
+        selectdatesString += '<option value="">Select Option</option>';
+        selectdatesString += '<option value="1">Day/Days</option>';
+        selectdatesString += '<option value="7">Week/Weeks</option>';
+        selectdatesString += '<option value="30">Month/Months</option>';
+        selectdatesString += "</select>";
+        selectdatesString += "<br>";
+        createElement("div", "date_select", "email_container");
+        document.getElementById("date_select").innerHTML = selectdatesString;
+    } else {
+        removeAllChildren("email_container");
+    }
+}
+function emailDateSelected() {
+    let dates_selected = document.getElementById("dates_selected").value;
+    let days_display = "";
+    switch (dates_selected) {
```

Comment:
Might want to consider converting this to an int and then all your cases would just be ints.